src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,37 +38,59 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
+
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        splitAndStoreAddress(trimmedAddress);
     }
 
     /**
      * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        String[] addressComponents = test.split(", ");
+
+        return addressComponents.length == 4;
     }
 
     @Override
     public String toString() {
-        return value;
+        String addressString = blockNumber.getBlockNumber() + ", "
```

Comment:
Perhaps we can improve this line by omitting `String addressString`?

Perhaps this?
``` java
    return blockNumber.getBlockNumber() + ", "
               .....
               .....
```